Assets/Scripts/SS3D/Core/Networking/PlayerControl/PlayerControlSystem.cs

```diff
@@ -0,0 +1,72 @@
+using System.Linq;
+using FishNet;
+using FishNet.Connection;
+using FishNet.Object;
+using FishNet.Object.Synchronizing;
+using SS3D.Core.Networking.PlayerControl.Messages;
+using SS3D.Core.Systems.Entities;
+using UnityEngine;
+
+namespace SS3D.Core.Networking.PlayerControl
+{
+    /// <summary>
+    /// Controls the player flux, when users want to authenticate, rejoin the game, leave the game
+    /// </summary>
+    public sealed class PlayerControlSystem : NetworkBehaviour
+    {
+        [SerializeField] private GameObject _soulPrefab;
+
+        [SyncObject]
+        private readonly SyncList<Soul> _serverSouls = new();
+
+        public string GetSoulCkeyByConn(NetworkConnection conn) => _serverSouls.Where(soul => soul.Owner == conn).ToArray()[0].Ckey;
+
+        private void Start()
+        {
+            SubscribeToEvents();
+        }
+
+        private void SubscribeToEvents()
+        {
+            InstanceFinder.ServerManager.RegisterBroadcast<UserAuthorizationMessage>(HandleAuthorizePlayer);
+        }
+
+        /// <summary>
+        /// Used by the server to validate credentials and reassign souls to clients.
+        /// TODO: Actual authentication
+        /// </summary>
+        /// <param name="conn">Network connection</param>
+        /// <param name="userAuthorizationMessage">struct containing the ckey and the connection that sent it</param>
+        [Server]
+        private void HandleAuthorizePlayer(NetworkConnection conn, UserAuthorizationMessage userAuthorizationMessage)
+        {
+            string ckey = userAuthorizationMessage.Ckey;
+
+            Soul match = null;
+            foreach (Soul soul in _serverSouls.Where((soul) => soul.Ckey == ckey))
```

Comment:
This will loop through every soul. Should it not try to find a single one or error when there are multiple?